### PR TITLE
potentially fix `import WalletConnect` issue

### PR DESCRIPTION
**Pull request by hamirmahal**

### CODE REVIEW

**Review by DanBurton — APPROVED** [review]

**Review by jeapostrophe — CHANGES_REQUESTED** [review]
If this works, then the examples in the docs and the patch in rps-9-web should be adjusted

**Comment by hamirmahal** [comment]
I can make those changes if this works.

At the time of this writing, there are [4 search results for `ALGO_WalletConnect`](https://github.com/reach-sh/reach-lang/search?q=ALGO_WalletConnect).

Where else would we need to make changes besides

1. [examples/rps-9-web/algoWC.patch](https://github.com/reach-sh/reach-lang/blob/0cf41a6b13b7b01dfa20abce62de3eb41c1f0970/examples/rps-9-web/algoWC.patch), 
2. [js/stdlib/ts/ALGO_WalletConnect.ts](https://github.com/reach-sh/reach-lang/blob/23f379d207b2938031c64ede14c63fe709598643/js/stdlib/ts/ALGO_WalletConnect.ts),
3. [docs/src/frontend/index.md](https://github.com/reach-sh/reach-lang/blob/1ee34d9292894f000124b1ab20eb3411c25286b8/docs/src/frontend/index.md), and
4. [docs/src/tut/rps/index.md](https://github.com/reach-sh/reach-lang/blob/0648dcb4d021084212b200140375baf53da2c5ed/docs/src/tut/rps/index.md)?

**Comment by jeapostrophe** [comment]
Use the grep, Hamir. Those are the only places I see when I do `ag ALGO_WalletConnect`

File: docs/src/frontend/index.md [hunk 1]
```diff
@@ -768,10 +768,11 @@ stdlib.setWalletFallback(stdlib.walletFallback({
   providerEnv: 'TestNet', MyAlgoConnect }));
 ```
 
-If the key `WalletConnect` is provided, and bound to the export of `@reach-sh/stdlib/ALGO_WalletConnect`, then [WalletConnect](https://walletconnect.com/) is used to connect to the [Algorand Wallet](https://algorandwallet.com/) for signing.
+If the key `WalletConnect` is provided, and bound to the `ALGO_WalletConnect` field of the export of `@reach-sh/stdlib`, then [WalletConnect](https://walletconnect.com/) is used to connect to the [Algorand Wallet](https://algorandwallet.com/) for signing.
```
Comment (jeapostrophe) [inline docs/src/frontend/index.md:771]: I'd change "field of the export of" to "export of"

File: docs/src/frontend/index.md [hunk 2]
```diff
@@ -768,10 +768,11 @@ stdlib.setWalletFallback(stdlib.walletFallback({
...
 For example, this sets the wallet fallback to be WalletConnect and the Algorand TestNet:
 ```js
-import WalletConnect from '@reach-sh/stdlib/ALGO_WalletConnect';
+import * as reachsdk from '@reach-sh/stdlib';
+const WalletConnect = reachsdk.ALGO_WalletConnect;
```
Comment (jeapostrophe) [inline docs/src/frontend/index.md:775]: I'd write `import { ALGO_WalletConnect as WalletConnect } from '@reach-sh/stdlib';`

File: docs/src/tut/rps/index.md [hunk 3]
```diff
@@ -1835,7 +1835,8 @@ stdlib.setWalletFallback(stdlib.walletFallback({
 
 Or, you could have your users use [WalletConnect](https://walletconnect.com/) to connect to the [Algorand Wallet](https://algorandwallet.com/), by adding the following:
 ```js
-import WalletConnect from '@reach-sh/stdlib/ALGO_WalletConnect';
+import * as reachsdk from '@reach-sh/stdlib';
+const WalletConnect = reachsdk.ALGO_WalletConnect;
```
Comment (jeapostrophe) [inline docs/src/tut/rps/index.md:1839]: See above

File: examples/rps-9-web/algoWC.patch [hunk 4]
```diff
@@ -4,7 +4,8 @@
  import './index.css';
  import * as backend from './build/index.main.mjs';
  import {loadStdlib} from '@reach-sh/stdlib';
-+import WalletConnect from '@reach-sh/stdlib/ALGO_WalletConnect';
++import * as reachsdk from '@reach-sh/stdlib';
++const WalletConnect = reachsdk.ALGO_WalletConnect;
```
Comment (jeapostrophe) [inline examples/rps-9-web/algoWC.patch:8]: See above

**Review by jeapostrophe — CHANGES_REQUESTED** [review]

File: docs/src/frontend/index.md (hunk 2 above)
Comment (hamirmahal) [reply docs/src/frontend/index.md]: [b0b8b5d](https://github.com/reach-sh/reach-lang/pull/664/commits/b0b8b5d7c8b6ac33868b17c5f5503189e1ea9701) addresses this.

File: docs/src/tut/rps/index.md (hunk 3 above)
Comment (hamirmahal) [reply docs/src/tut/rps/index.md]: [b0b8b5d](https://github.com/reach-sh/reach-lang/pull/664/commits/b0b8b5d7c8b6ac33868b17c5f5503189e1ea9701) addresses this.

File: examples/rps-9-web/algoWC.patch (hunk 4 above)
Comment (hamirmahal) [reply examples/rps-9-web/algoWC.patch]: [b0b8b5d](https://github.com/reach-sh/reach-lang/pull/664/commits/b0b8b5d7c8b6ac33868b17c5f5503189e1ea9701) addresses this.

**Review by jeapostrophe — COMMENTED** [review]
Okay, this looks good. I want to rely on @DanBurton to merge though, so he can check that it works in his testing environment.

BTW, we want to do the same thing with MyAlgo

**Review by DanBurton — CHANGES_REQUESTED** [review]
When I run `examples/rps-9-web` (patched w/ `algoWC.patch`) like this:

```
(cd ../../js/ && make build) && REACH_CONNECTOR_MODE=ALGO ../../reach react
```

I'm seeing this error:

```
TypeError: t is not a constructor
```

I have thoughts about how to fix this but I'm going to work on a release first and come back to this later to day.

**Comment by DanBurton** [comment]
Here are both my practical and my stylistic recommendations. I think `index.ts` should do this:

```
import MyAlgoConnect from './ALGO_MyAlgoConnect';
import WalletConnect from './ALGO_WalletConnect';

export const wallets = {
  ALGO: {
    MyAlgoConnect,
    WalletConnect,
  },
};
```

And then usage of it looks something like this:

```
import { loadStdlib, wallets } from '@reach-sh/stdlib';
const reach = loadStdlib('ALGO');
const { MyAlgoConnect } = wallets.ALGO;
reach.setWalletFallback(reach.walletFallback({MyAlgoConnect}));
```

The problem `t is not a constructor` happens because of `import * as ALGO_MyAlgoConnect` instead of `import MyAlgoConnect`. The latter style uses the default export, rather than grouping all exports into an object. See the `export default` statements in those respective files being imported.

**Comment by jeapostrophe** [comment]
Do it! Ship it!

**Comment by DanBurton** [comment]
I have tested that my proposed change works. There are some other issues w/ the WalletConnect impl that we have that arise later, but they are unrelated to this change.

**Comment by jeapostrophe** [comment]
Great! We'll have to update the MAC docs too

File: docs/src/frontend/index.md [hunk 5]
```diff
@@ -760,20 +760,28 @@ On Algorand, it can provide a wallet that directly connects to the Algorand netw
...
+const reach = loadStdlib('ALGO');
+const { MyAlgoConnect } = wallets.ALGO;
+reach.setWalletFallback(reach.walletFallback({
+  MyAlgoConnect,
+  providerEnv: 'TestNet',
+}));
 ```
 
-If the key `WalletConnect` is provided, and bound to the export of `@reach-sh/stdlib/ALGO_WalletConnect`, then [WalletConnect](https://walletconnect.com/) is used to connect to the [Algorand Wallet](https://algorandwallet.com/) for signing.
+If the key `WalletConnect` is provided, and bound to the `wallets.ALGO` export of `@reach-sh/stdlib`, then [WalletConnect](https://walletconnect.com/) is used to connect to the [Algorand Wallet](https://algorandwallet.com/) for signing.
```
Comment (DanBurton) [inline docs/src/frontend/index.md:775]: "... and bound to the `wallets.ALGO.WalletConnect` export ..."

File: docs/src/frontend/index.md [hunk 6]
```diff
@@ -760,20 +760,28 @@ On Algorand, it can provide a wallet that directly connects to the Algorand netw
 The network connection is specified via the `providerEnv` key, which may be a string (which is used as an argument to `{!js} providerEnvByName`) or an environment (which is used as an argument to `{!js} setProviderByEnv`).
 By default, signing is via an interactive browser window prompt, where the user repeatedly provides their mnemonic.
 
-If the key `MyAlgoConnect` is provided, and bound to the export of `@reach-sh/stdlib/ALGO_MyAlgoConnect`, then [My Algo](https://wallet.myalgo.com/home) will be used for signing.
+If the key `MyAlgoConnect` is provided, and bound to an export of `@reach-sh/stdlib`, then [My Algo](https://wallet.myalgo.com/home) will be used for signing.
```
Comment (DanBurton) [inline docs/src/frontend/index.md:763]: "... and bound to the `wallets.ALGO.MyAlgoConnect` export ..."

Comment (hamirmahal) [reply docs/src/frontend/index.md]: [bc89dab](https://github.com/reach-sh/reach-lang/pull/664/commits/bc89dabffbb9fbbd23289ff8313f8dc1723415a0) should address this.

File: docs/src/frontend/index.md (hunk 5 above)
Comment (hamirmahal) [reply docs/src/frontend/index.md]: [bc89dab](https://github.com/reach-sh/reach-lang/pull/664/commits/bc89dabffbb9fbbd23289ff8313f8dc1723415a0) should address this.

Comment (hamirmahal) [reply docs/src/frontend/index.md]: [e7a6dc8](https://github.com/reach-sh/reach-lang/pull/664/commits/e7a6dc85c8039351f86a3b2ed7dd2312371259c8) should address this.

File: docs/src/frontend/index.md (hunk 6 above)
Comment (hamirmahal) [reply docs/src/frontend/index.md]: [e7a6dc8](https://github.com/reach-sh/reach-lang/pull/664/commits/e7a6dc85c8039351f86a3b2ed7dd2312371259c8) should address this.

**Review by DanBurton — DISMISSED** [review]

**Comment by lorenzopolidori** [comment]
Can this PR potentially fix this issue? https://github.com/reach-sh/reach-lang/issues/710

**Comment by jeapostrophe** [comment]
Yes @lorenzopolidori that's the idea... we're just waiting for it to be finalzied

**Review by jeapostrophe — APPROVED** [review]

**Pull request merged**